grails-app/controllers/org/pih/warehouse/api/PicklistApiController.groovy

```diff
@@ -0,0 +1,14 @@
+package org.pih.warehouse.api
+
+import org.pih.warehouse.picklist.PicklistService
+
+class PicklistApiController extends BaseDomainApiController {
+
+    PicklistService picklistService
+
+    def clearPicklist() {
+        picklistService.clearPicklist(params.id)
+
+        render status: 204
```

Comment:
Here's a stackoverflow about deleting "some" of a resource collection, but it doesn't discuss response codes. 
https://stackoverflow.com/questions/37138720/restful-api-delete-some-of-a-resource-collection

If anyone finds good documentation on response codes for deleting a resource collection and in particular  first vs subsequent requests with different results i.e. deleted X items vs. resource collection has no items vs deleted 0 items because there were filters applied to the nested resource collection

* /api/picklists/:id/items
* /api/picklists/:id/items
* /api/picklists/:id/items?filter=value

Reply:
@jmiranda Definitely topic worth to extend, but from my understanding, since we do not return anything in the response, we should be fine with the `204`, since this is "No Content", even though we delete a collection.
I've always said I was "pro" a silent delete - so that e.g. even if there is nothing to delete or a resource of given id is not found, we still return 204.
This is about idempotency of HTTP methods. Looking at many resources, it feels like DELETE is idempotent, hence it should return the same response for a single request even if called many times, so in our case - even if we attempt to delete an empty collection, we should still get 204.
https://stackoverflow.com/a/6440374/23546798

> So regardless of whether you DELETE an existing resource, or attempt to DELETE a resource that does not exist, the server resource state is the same.